xarray/tests/duckarrays/base/strategies.py

```diff
@@ -75,17 +80,14 @@ def dataset(
     max_vars=5,
 ):
     names = st.text(min_size=1)
-    sizes = st.dictionaries(
-        keys=names,
-        values=st.integers(min_value=min_size, max_value=max_size),
-        min_size=min_dims,
-        max_size=max_dims,
+    sizes = dimension_sizes(
+        min_size=min_size, max_size=max_size, min_dims=min_dims, max_dims=max_dims
     )
 
     data_vars = sizes.flatmap(
         lambda s: st.dictionaries(
             keys=names.filter(lambda n: n not in s),
```

Comment:
This filter will need to be changed to reflect the new type of s.

Reply:
thanks, I forgot running all related tests before pushing.